src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.Impl.cs

```diff
@@ -703,6 +754,84 @@ private static bool CanTypeBeExtendedPubliclyImpl(ITypeSymbol type)
                         m => m.DeclaredAccessibility != Accessibility.Internal && m.DeclaredAccessibility != Accessibility.Private && m.DeclaredAccessibility != Accessibility.ProtectedAndInternal
                     );
             }
+
+            /// <summary>
+            /// Various Visit* methods return true if an oblivious reference type is detected.
+            /// </summary>
+            private sealed class ObliviousDetector : SymbolVisitor<bool>
+            {
+                private bool _isTypeDeclaration = true;
+
+                public override bool VisitField(IFieldSymbol symbol)
+                {
+                    _isTypeDeclaration = false;
+                    return Visit(symbol.Type);
+                }
+
+                public override bool VisitMethod(IMethodSymbol symbol)
```

Comment:
❔ What about oblivious types in generic type constraints?

Reply:
Ah, I think that's probably one of the cases Jared had in mind (`where T : class~` vs. `where T : class!` or `where T : class?` kinda situations). I'll address